src/main/java/mate/academy/springboot/datajpa/mapper/ProductMapper.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.springboot.datajpa.mapper;
+
+import mate.academy.springboot.datajpa.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductMapper implements RequestDtoMapper<ProductRequestDto, Product>,
+        ResponseDtoMapper<ProductResponseDto, Product> {
+    private final CategoryService categoryService;
+
+    public ProductMapper(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    @Override
+    public Product mapToModel(ProductRequestDto dto) {
+        Product product = new Product();
+        product.setPrice(dto.getPrice());
+        product.setTitle(dto.getTitle());
+        product.setCategory(categoryService.getById(dto.getCategoryId()));
+        return product;
+    }
+
+    @Override
+    public ProductResponseDto mapToDto(Product product) {
+        ProductResponseDto responseProductDto = new ProductResponseDto();
```

Comment:
```suggestion
        ProductResponseDto dto = new ProductResponseDto();
```
be consistent with your other code